templates/kubernetes-cluster.template

```diff
@@ -658,16 +668,21 @@ Resources:
               mode: '000755'
             "/etc/systemd/system/awslogs.service":
               source: !Sub "https://${QSS3BucketName}.s3.amazonaws.com/${QSS3KeyPrefix}/scripts/awslogs.service"
+            "/tmp/setup-kubelet-hostname.sh":
+              source: !Sub "https://${QSS3BucketName}.s3.amazonaws.com/${QSS3KeyPrefix}/scripts/setup-kubelet-hostname.sh"
+              mode: '000755'
           commands:
             "00-kubernetes-override-binaries":
               command: "/tmp/kubernetes-override-binaries.sh"
             "01-cloudwatch-agent-setup":
               command: !Sub "python /usr/local/aws/awslogs-agent-setup.py -n -r ${AWS::Region} -c /tmp/kubernetes-awslogs.conf"
             "02-cloudwatch-service-config":
               command: "systemctl enable awslogs.service && systemctl start awslogs.service"
+            "03-setup-kubelet-hostname":
+              command: "/tmp/setup-kubelet-hostname.sh"
             # This joins the existing cluster with kubeadm.  Kubeadm is preinstalled in the AMI this template uses, so
             # all that's left is to join the cluster with the correct token.
-            "03-k8s-setup-node":
+            "04-k8s-setup-node":
               command: !Sub "kubeadm reset && kubeadm join --node-name=\"$(hostname -f)\" --token=${KubeadmToken.Token} ${K8sMasterInstance.PrivateIp}:6443"
```

Comment:
`kubeadm join` won't work unchanged. With 1.9.x we need to validate certs. I'd recommend using `--discovery-token-unsafe-skip-ca-verification` for now until we have a better solution.